x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/helpers.ts

```diff
@@ -78,13 +89,20 @@ export const streamGraph = async ({
   };
 
   let finalMessage = '';
-  const stream = assistantGraph.streamEvents(inputs, {
-    callbacks: [apmTracer, ...(traceOptions?.tracers ?? [])],
-    runName: DEFAULT_ASSISTANT_GRAPH_ID,
-    streamMode: 'values',
-    tags: traceOptions?.tags ?? [],
-    version: 'v1',
-  });
+  let conversationId: string | undefined;
+  const stream = assistantGraph
+    .withListeners({
+      onEnd: (run: Run) => {
+        conversationId = handleEndRun(run);
+      },
+    })
```

Comment:
Why a `withListeners` callback vs having persistence be the final node in the graph?